mimium-lang/src/runtime/vm.rs

```diff
@@ -76,33 +71,57 @@ impl StateStorageStack {
     }
 }
 
+// Upvalues are used with Rc<RefCell<UpValue>> because it maybe shared between multiple closures
+// Maybe it will be managed with some GC mechanism in the future.
 #[derive(Debug, Clone, PartialEq)]
-pub enum UpValue {
+enum UpValue {
     Open(OpenUpValue),
-    Closed(Rc<RefCell<Vec<RawVal>>>),
+    Closed(Vec<RawVal>),
 }
+type SharedUpValue = Rc<RefCell<UpValue>>;
 impl From<OpenUpValue> for UpValue {
     fn from(value: OpenUpValue) -> Self {
         Self::Open(value)
     }
 }
 
+#[derive(Default)]
+struct LocalUpValueMap(Vec<(Reg, SharedUpValue)>);
```

Comment:
This `LocalUpValueMap` is equivalent to the `OpenBroker` in the Lua Interpreter with Rust Tutorial.

https://github.com/WuBingzheng/build-lua-in-rust/blob/main/listing/ch09.closure/src/vm.rs#L85

They use binary search for shared-value search but I used iterator::find because the number of upvalues are usually small(need detailed comparison with the benchmark)